### ISSUE TITLE
package:analyzer version >=3.3.0 seems to have issues with version-specific language features

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/sdk#48499]
Related to an issue that was raised in one of my projects: https://github.com/rrousselGit/freezed/issues/611

It seems that with analyzer 3.3.0, it fails to analyze code that is using generic annotations in a codebase that is using Dart >=2.14

This causes the following error:

```
[SEVERE] freezed:freezed on test/integration/implements_decorator.dart:
An error `FormatterException` occurred while formatting the generated source for
  `asset:freezed/test/integration/implements_decorator.dart`
which was output to
  `test/integration/implements_decorator.freezed.dart`.
This may indicate an issue in the generator, the input source code, or in the
source formatter.
Could not format because the source could not be parsed:

line 430, column 6 of .: This requires the 'generic-metadata' language feature to be enabled.
    ╷
430 │ @With<House>()
    │      ^
```

when an experimental flag shouldn't be necessary as the pubspec contains

```yaml
sdk: ">=2.14.0"
```

which has the feature enabled by default.

Changing the analyzer version to use `analyzer: ">=3.0.0 <3.3.0"` makes the error disappear, which seems to indicate that there is a regression in the analyzer.

**Comment by rrousselGit** [comment]
Actually this may be caused by dart_style – which depends on the analyzer

Although I'm not sure why a new version of the analyzer broke dart_style in such a way

**Comment by SunlightBro** [comment]
Some additional Info
dart_style 2.2.1 uses this [feature set](https://github.com/dart-lang/dart_style/blob/610ef5797764deb269e24653454886047fe6dc56/lib/src/dart_formatter.dart#L90)
```dart
var featureSet = FeatureSet.fromEnableFlags2(
    sdkLanguageVersion: Version(2, 13, 0),
    flags: [
      'constructor-tearoffs',
      'generic-metadata',
      'nonfunction-type-aliases',
      'triple-shift'
    ]);
```
This is probably why source_gen fails.

**Comment by rrousselGit** [comment]
> Which version of dart_style do you use?


Latest/2.2.1.

**Comment by scheglov** [comment]
I recommend you to request publishing the new `dart_style` in their repository.

**Comment by munificent** [comment]
I've published 2.2.2 which should have the fixes in.
